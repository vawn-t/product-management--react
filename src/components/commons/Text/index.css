/*------------------------------------*\
# TEXT
\*------------------------------------*/
.text-default {
  color: var(--dark);
  letter-spacing: 0.04rem;
  font-size: 16px;
}

.text-highlight { 
  color: var(--secondary);
  letter-spacing: 0.04rem;
  text-transform: uppercase;
  font-size: 16px;
}
