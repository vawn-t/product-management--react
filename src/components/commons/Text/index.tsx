// Library
import React, { memo, ReactNode } from 'react';

// Types
import { FsType, FwType, VariantsTypes } from '@common-types/index';

// Styles
import './index.css';

interface TextProps {
  children: ReactNode;
  color?: string;
  fontStyle?: FsType;
  fontWeight?: FwType;
  fontSize?: string;
  variant?: VariantsTypes;
}

const Text: React.FC<TextProps> = ({
  children,
  color,
  fontWeight = FwType.Normal,
  fontSize,
  fontStyle = FsType.Inherit,
  variant = VariantsTypes.Default
}) => (
  <p
    className={`text-${variant}`}
    style={{
      fontSize: fontSize,
      color: color,
      fontStyle: fontStyle,
      fontWeight: fontWeight
    }}
  >
    {children}
  </p>
);

export default memo(Text);
