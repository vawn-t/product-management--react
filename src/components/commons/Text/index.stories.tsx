// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import Text from './index';

// Types
import { FwType, VariantsTypes } from '@common-types/index';

// Styles
import '@assets/styles/reset.css';
import '@assets/styles/App.css';
import '@assets/styles/variables.css';

export default {
  title: 'Components/Text',
  component: Text
} as ComponentMeta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args) => <Text {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Text'
};

export const Highlight = Template.bind({});
Highlight.args = {
  children: 'Text',
  variant: VariantsTypes.Highlight,
  fontWeight: FwType.Bold
};
