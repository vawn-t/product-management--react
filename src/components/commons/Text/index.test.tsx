// Libraries
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';

// Components
import Text from './index';

// Types
import { FsType, FwType, VariantsTypes } from '@common-types/index';

describe('Button component', () => {
  let container: HTMLElement;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    container.remove();
    jest.clearAllMocks();
    cleanup();
  });

  test('should render correctly', () => {
    const defaultText = create(<Text>text</Text>).toJSON();
    expect(defaultText).toMatchSnapshot();

    const highlightText = create(
      <Text variant={VariantsTypes.Highlight}>Typography</Text>
    ).toJSON();
    expect(highlightText).toMatchSnapshot();

    const boldText = create(
      <Text fontWeight={FwType.Bold}>Typography</Text>
    ).toJSON();
    expect(boldText).toMatchSnapshot();

    const italicText = create(
      <Text fontStyle={FsType.Italic}>Typography</Text>
    ).toJSON();
    expect(italicText).toMatchSnapshot();
  });

  test('should render Text with content is Typography', () => {
    const text = 'Typography';

    const { getByText } = render(<Text>{text}</Text>, { container });

    expect(getByText(text).textContent).toMatch(text);
  });
});
