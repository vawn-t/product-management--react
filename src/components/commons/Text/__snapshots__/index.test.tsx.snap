// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<p
  className="text-default"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "inherit",
      "fontWeight": 400,
    }
  }
>
  text
</p>
`;

exports[`Button component should render correctly 2`] = `
<p
  className="text-highlight"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "inherit",
      "fontWeight": 400,
    }
  }
>
  Typography
</p>
`;

exports[`Button component should render correctly 3`] = `
<p
  className="text-default"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "inherit",
      "fontWeight": 700,
    }
  }
>
  Typography
</p>
`;

exports[`Button component should render correctly 4`] = `
<p
  className="text-default"
  style={
    Object {
      "color": undefined,
      "fontSize": undefined,
      "fontStyle": "italic",
      "fontWeight": 400,
    }
  }
>
  Typography
</p>
`;
